src/ray/object_manager/plasma/client.cc

```diff
@@ -569,6 +693,62 @@ Status PlasmaClient::Impl::Get(const std::vector<ObjectID> &object_ids,
       &object_ids[0], num_objects, timeout_ms, wrap_buffer, &(*out)[0], is_from_worker);
 }
 
+void PlasmaClient::Impl::EnsureGetAcquired(
+    std::unique_ptr<ObjectInUseEntry> &object_entry) {
+  PlasmaObject *object = &object_entry->object;
+  auto plasma_header = GetPlasmaObjectHeader(*object);
+  if (object_entry->read_acquired) {
+    return;
+  }
+
+  int64_t version_read = plasma_header->ReadAcquire(object_entry->next_version_to_read);
+  object_entry->read_acquired = true;
+  if (version_read > 0) {
+    object_entry->is_mutable = true;
+    object_entry->next_version_to_read = version_read;
+
+    // The data and metadata size may have changed, so update here before we
+    // create the Get buffer to return.
+    object_entry->object.data_size = plasma_header->data_size;
```

Comment:
we don't handle different data size in this PR yet right?

Reply:
Hmm looking at our impl, I am not sure if we can handle write in the driver and read in the worker with different size.

Can you try adding a test like this? (it is what I wrote in the dag branch)

```python3
"""
Verify put in 2 different processes work.
"""
ray.init(num_cpus=1)

print("Test Write input from driver -> Read & Write from worker -> Read output from driver")
expected_input = b"000000000000"
ref = ray.put(expected_input, max_readers=1)
print(ref)

@ray.remote
class A:
    def f(self, refs, expected_input, output_val):
        ref = refs[0]
        val = ray.get(refs[0])
        assert val == expected_input, val
        ray.release(ref)
        ray.worker.global_worker.put_object(output_val, object_ref=ref, max_readers=1)


a = A.remote()
time.sleep(1)
output_val = b"0"
b = a.f.remote([ref], expected_input, output_val)
ray.get(b)
val = ray.get(ref)
assert output_val == val
ray.release(ref)

print("Test Write input from driver twice -> Read & Write from worker -> Read output from driver")
# Test write twice.
ref = ray.put(b"000000000000", max_readers=1)
assert b"000000000000" == ray.get(ref)
ray.release(ref)
print(ref)
expected_input = b"1"
ray.worker.global_worker.put_object(b"1", object_ref=ref, max_readers=1)

a = A.remote()
time.sleep(1)
expected_output = b"23"
b = a.f.remote([ref], expected_input, expected_output)
ray.get(b)
val = ray.get(ref)
assert expected_output == val
ray.release(ref)
```

Reply:
It does work, but the test is a good idea. Added.